bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/rewriter/gemini/GeminiRewriter.java

```diff
@@ -33,29 +33,32 @@
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class GeminiRewriter {
 
     private static final Logger logger = LoggerFactory.getLogger(GeminiRewriter.class);
-    private static final String API_URI_TEMPLATE = "https://us-central1-aiplatform.googleapis.com/v1/projects/%s/locations/us-central1/publishers/google/models/gemini-1.0-pro:streamGenerateContent?alt=sse";
-
-    public static void rewriteSQL(InputQuery inputQuery, List<AntiPatternVisitor> visitorsThatFoundAntiPatterns,
-                                  String projectId, PromptYamlReader promptYamlReader) throws IOException {
-        try {
-            String queryStr = inputQuery.getQuery();
-            for (AntiPatternVisitor visitor : visitorsThatFoundAntiPatterns) {
-                String prompt = promptYamlReader.getAntiPatternNameToPrompt().get(visitor.getName());
-                if (prompt != null) {
-                    prompt = String.format(prompt, queryStr).replace("%%","%");
-                    queryStr = processPrompt(prompt, projectId);
-                }
-            }
-            if (!queryStr.equals(inputQuery.getQuery())) {
-                inputQuery.setOptimizedQuery(queryStr);
+    private static final String API_URI_TEMPLATE = "https://us-central1-aiplatform.googleapis.com/v1/projects/%s/locations/us-central1/publishers/google/models/gemini-1.5-pro-preview-0409:generateContent?alt=sse";
```

Comment:
is the preview GA?

Reply:
Let me double check if it's not. It could be a good idea to set a parameter to set this model as well.

Reply:
It is in fact GA: [link](https://cloud.google.com/vertex-ai/generative-ai/docs/learn/models?_gl=1*l4sq94*_ga*MTIwNzM5MzAzNS4xNzA3ODY0ODQ1*_ga_WH2QY8WWF5*MTcxNDA1MzI4Mi44MS4xLjE3MTQwNTMzMDQuMC4wLjA.&_ga=2.210488725.-1207393035.1707864845&_gac=1.88288745.1713364073.CjwKCAjw5v2wBhBrEiwAXDDoJWucTYLLISNkmPtAJcDqccFdR0w-jzlQmR7XpsDYqRMqo2u1Ck2-PBoC1xUQAvD_BwE)